src/StepperFirmata2.cpp

```diff
@@ -0,0 +1,401 @@
+        } else if (wireCount == 3 && stepType == 0) {
+          stepper[deviceNum] = new AccelStepper(AccelStepper::FULL3WIRE, directionPin, stepPin, motorPin3);
+        } else if (wireCount == 3 && stepType == 2) {
+          stepper[deviceNum] = new AccelStepper(AccelStepper::HALF3WIRE, directionPin, stepPin, motorPin3);
+        } else if (wireCount == 4 && stepType == 0) {
+          stepper[deviceNum] = new AccelStepper(AccelStepper::FULL4WIRE, directionPin, stepPin, motorPin3, motorPin4, false);
+        } else if (wireCount == 4 && stepType == 2) {
+          stepper[deviceNum] = new AccelStepper(AccelStepper::HALF4WIRE, directionPin, stepPin, motorPin3, motorPin4);
+        }
+
+        if (argc >= index) {
+          invertPins = argv[index]; // optional byte 11, 12 or 13
+          if (wireCount == 0) {
+             stepper[deviceNum]->setPinsInverted(invertPins & 0x01, invertPins >> 1 & 0x01, invertPins >> 4 & 0x01);
+          } else {
+            stepper[deviceNum]->setPinsInverted(invertPins & 0x01, invertPins >> 1 & 0x01, invertPins >> 2 & 0x01, invertPins >> 3 & 0x01, invertPins >> 4 & 0x01);
+          }
+        }
+
+        if (interface & 0x01) {
+          stepper[deviceNum]->setEnablePin(enablePin);
+        }
+
+        isRunning[deviceNum] = false;
+
+      }
+
+      else if (stepCommand == STEPPER_STEP) {
+
+        numSteps = decode32BitSignedInteger(argv[2], argv[3], argv[4], argv[5], argv[6]);
+
+        if (stepper[deviceNum]) {
+          stepper[deviceNum]->move(numSteps);
+          isRunning[deviceNum] = true;
+        }
+
+      }
+
+      else if (stepCommand == STEPPER_ZERO) {
+        if (stepper[deviceNum]) {
+          stepper[deviceNum]->setCurrentPosition(0);
+        }
+      }
+
+      else if (stepCommand == STEPPER_TO) {
+        if (stepper[deviceNum]) {
+          numSteps = decode32BitSignedInteger(argv[2], argv[3], argv[4], argv[5], argv[6]);
+          stepper[deviceNum]->moveTo(numSteps);
+          isRunning[deviceNum] = true;
+        }
+      }
+
+      else if (stepCommand == STEPPER_ENABLE) {
+        if (stepper[deviceNum]) {
+          if (argv[2] == 0x00) {
+            stepper[deviceNum]->disableOutputs();
+          } else {
+            stepper[deviceNum]->enableOutputs();
+          }
+        }
+      }
+
+      else if (stepCommand == STEPPER_STOP) {
+        if (stepper[deviceNum]) {
+          stepper[deviceNum]->stop();
+          isRunning[deviceNum] = false;
+          reportPosition(deviceNum, true);
+        }
+      }
+
+      else if (stepCommand == STEPPER_REPORT_POSITION) {
+        if (stepper[deviceNum]) {
+          reportPosition(deviceNum, false);
+        }
+      }
+
+      else if (stepCommand == STEPPER_SET_ACCELERATION) {
+
+        float decodedAcceleration = decodeCustomFloat(argv[2], argv[3], argv[4], argv[5]);
+
+        if (stepper[deviceNum]) {
+          if (decodedAcceleration == 0.0) {
+            stepper[deviceNum]->setAcceleration(99999);
```

Comment:
I see, you've come up with a clever hack here! I ran some tests to get a sense for how this affects accuracy vs using setSpeed directly. It's not too bad. Just off by a few milliseconds. Here's what I did (and then run with USE_HACK true, then false and compare output):

```c++
#include <AccelStepper.h>

#define MAX_ACCELERATION 99999
#define USE_HACK true

// params: step, direction
AccelStepper stepper(AccelStepper::DRIVER, 3, 2);

long numSteps = 1000 * 8; // 1000 steps
long startTime;
bool runOnce = true;

void setup() {
  Serial.begin(9600);

  if (USE_HACK) {
    stepper.setMaxSpeed(500);
    stepper.move(numSteps);
    stepper.setAcceleration(MAX_ACCELERATION);
  } else {
    stepper.setMaxSpeed(3000);
    stepper.move(numSteps);
    stepper.setSpeed(500);    
  }

   startTime = millis();
}

void loop() {
  if (USE_HACK) { 
    stepper.run();
  } else {
    stepper.runSpeedToPosition();
  }
       
  if (runOnce && stepper.currentPosition() == numSteps) {
    Serial.println(millis() - startTime);
    runOnce = false;
  }
}
```

I found that using a lower number than 99999 improves accuracy, but I'm not sure if that is always the case so 99999 is probably safe, but please add a constant `MAX_ACCELERATION` to define it. Also add a comment about how this approach is a hack.